### ISSUE TITLE
Not working ^ 

### ISSUE THREAD

**Issue by bilelmoussaoui** [alexhuntley/Plots#1]
Coming from https://github.com/flathub/flathub/pull/2036

I still can't make ^ work properly,  the issue is probably related to my locale 
```
bilelmoussaoui@fedora ~ > locale 
LANG=en_US.UTF-8
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=
```

That's all I get from the terminal
```
bilelmoussaoui@fedora ~/P/contrast (wip/cdavis/gtk4) [SIGINT]> flatpak run com.github.alexhuntley.Plots
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 264, in on_pointer_move
    self.cursor.mouse_select(element, direction, drag=True)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 341, in mouse_select
    self.selection_bounds, self.selection_ancestor = self.calculate_selection()
TypeError: cannot unpack non-iterable NoneType object
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 147, in do_draw_cb
    self.expr.draw(ctx, self.cursor, widget_transform)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 652, in draw
    super().draw(ctx, cursor, widget_transform)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 551, in draw
    self.index_in_parent in cursor.selection_bounds:
TypeError: argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 147, in do_draw_cb
    self.expr.draw(ctx, self.cursor, widget_transform)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 652, in draw
    super().draw(ctx, cursor, widget_transform)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 551, in draw
    self.index_in_parent in cursor.selection_bounds:
TypeError: argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 264, in on_pointer_move
    self.cursor.mouse_select(element, direction, drag=True)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 341, in mouse_select
    self.selection_bounds, self.selection_ancestor = self.calculate_selection()
TypeError: cannot unpack non-iterable NoneType object
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 264, in on_pointer_move
    self.cursor.mouse_select(element, direction, drag=True)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 341, in mouse_select
    self.selection_bounds, self.selection_ancestor = self.calculate_selection()
TypeError: cannot unpack non-iterable NoneType object
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 264, in on_pointer_move
    self.cursor.mouse_select(element, direction, drag=True)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 341, in mouse_select
    self.selection_bounds, self.selection_ancestor = self.calculate_selection()
TypeError: cannot unpack non-iterable NoneType object
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 147, in do_draw_cb
    self.expr.draw(ctx, self.cursor, widget_transform)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 652, in draw
    super().draw(ctx, cursor, widget_transform)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 551, in draw
    self.index_in_parent in cursor.selection_bounds:
TypeError: argument of type 'NoneType' is not iterable
Traceback (most recent call last):
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 147, in do_draw_cb
    self.expr.draw(ctx, self.cursor, widget_transform)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 652, in draw
    super().draw(ctx, cursor, widget_transform)
  File "/app/lib/python3.8/site-packages/plots/formula.py", line 551, in draw
    self.index_in_parent in cursor.selection_bounds:
TypeError: argument of type 'NoneType' is not iterable
Shader compile failure (0): b"0:74(9): error: `s' undeclared\n0:74(2): error: `return' with wrong type error, in function `formula0' returning float\n"
Shader compile failure (0): b"0:74(9): error: `s' undeclared\n0:74(11): error: `i' undeclared\n0:74(9): error: operands to arithmetic operators must be numeric\n0:74(2): error: `return' with wrong type error, in function `formula0' returning float\n"
Shader compile failure (0): b"0:74(9): error: `sin' undeclared\n0:74(2): error: `return' with wrong type error, in function `formula0' returning float\n"
```

I think it could need some "Help page" with the list of operations you can use and how you can type them :)

**Comment by alexhuntley** [comment]
Thanks for reporting this. Which keyboard layout are you using? And what are you doing to get the TypeError exceptions?

**Comment by alexhuntley** [comment]
> I think it could need some "Help page" with the list of operations you can use and how you can type them :)

I'll open another issue :)

**Comment by bilelmoussaoui** [comment]
> Thanks for reporting this. Which keyboard layout are you using? And what are you doing to get the TypeError exceptions?

Opening the app and writing something on the equation field. My keyboard layout is BE_fr

**Comment by alexhuntley** [comment]
Could you checkout branch bug-1, run 
`python3 -m plots`
from the repo root, type in ^ and paste the output please?

**Comment by alexhuntley** [comment]
Just thinking, this issue is probably because circumflex is a dead key in AZERTY layouts

**Comment by bilelmoussaoui** [comment]
> Just thinking, this issue is probably because circumflex is a dead key in AZERTY layouts

That's probably the cause indeed.

**Comment by alexhuntley** [comment]
Potentially fixed, could you try the latest commit on branch bug-1 please?

**Comment by bilelmoussaoui** [comment]
Can confirm it fixes my issue, thanks!

**Comment by alexhuntley** [comment]
Fantastic! The fix is in v0.1.4, which should be on Flathub soon.
